sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFluxCore.java

```diff
@@ -255,7 +255,8 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
     private Flux<P> byPage(Supplier<PageRetriever<C, P>> provider, C continuationToken, Integer pageSize) {
         return Flux.defer(() -> {
             final PageRetriever<C, P> pageRetriever = provider.get();
-            final ContinuationState<C> state = new ContinuationState<>(continuationToken);
+            final ContinuationState<C> state = new ContinuationState<>(continuationToken,
+                getContinuationPredicate().negate());
```

Comment:
Should we use `hasNextPredicate` instead of `isDonePredicate` to avoid `.negate()` and also align with Java `Iterator`?

Reply:
That's a good idea, it'll also help reduce the chances of an accidental boolean inversion.